### PR TITLE
Add p6-tabs component

### PR DESCRIPTION
**Pull request by vuongc**

### CODE REVIEW

File: src/components/molecules/p6-tabs/p6-tabs.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Story, Preview, Meta } from "@storybook/addon-docs/blocks";
+import { action } from "@storybook/addon-actions";
+import readme from "./readme.md";
+
+<Meta title="Molecules/Tabs" component="p6-tabs" parameters={{ notes: readme }} />
+
+# Link
```
Comment (jswale) [inline src/components/molecules/p6-tabs/p6-tabs.stories.mdx:7]: Should be Tabs

File: src/components/molecules/p6-tabs/p6-tabs.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+`p6-tabs` provides a tabs component which display content based on the tab selected.
+
```
Comment (jswale) [inline src/components/molecules/p6-tabs/p6-tabs.stories.mdx:10]: Missing preview section

File: src/components/molecules/p6-tabs/p6-tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    const { id } = event.currentTarget as HTMLAnchorElement;
+    this.tabSelected = id;
+  };
+
+  /**
+   * Get tabs from slot sttributes.
+   */
+  private getTabs(): Tab[] {
+    return Array.from(this.host.children)
+      .filter(
```
Comment (jswale) [inline src/components/molecules/p6-tabs/p6-tabs.tsx:73]: This filter could be mutualized and also used inside the **getContent**

File: src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      <p6-tabs default="first">
+        <div id="first" title="What is Lorem Ipsum?">1</div>
+        <div id="second" title="Where does it come from?">2</div>
+        <div id="third" title="Why do we use it?">3</div>
+        <div id="fourth" title="Where can I get some?">4</div>
+      </p6-tabs>
+      `,
+    });
+
+    expect(page.root).toEqualHtml(`
```
Comment (jswale) [inline src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx:32]: You can use snapshot for that

File: src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      <p6-tabs>
+        <mock:shadow-root>
+         </mock:shadow-root>
+      </p6-tabs>
+    `);
+  });
+
+  it("Should display the first tab", async () => {
```
Comment (jswale) [inline src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx:19]: Add a new test setting the default tab to something different than the first one

File: src/components/molecules/p6-tabs/p6-tabs.stories.mdx (hunk 1 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.stories.mdx]: updated

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 3 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: It's not the same condition for the filter, in the second one you only get the one that is selected

File: src/components/molecules/p6-tabs/p6-tabs.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+   * Use this method by keeping a ref of the component.
+   *
+   * @param id Id to be selected
+   */
+  @Method()
+  async setTabSelected(id: string): Promise<void> {
+    if (this.getTabs().some((tab) => tab.id === id)) {
+      this.tabSelected = this.getTabId(id);
+    }
+  }
```
Comment (vincentv) [inline src/components/molecules/p6-tabs/p6-tabs.tsx:46]: Do we need this method? We could replace the **default** prop with a **selected** mutable prop.

The methods are not encouraged in the stencil doc.

> Developers should try to rely on publicly exposed methods as little as possible, and instead default to using properties and events as much as possible. As an app scales, we've found it's easier to manage and pass data through @Prop rather than public methods.

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 3 above)
Comment (jswale) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: I know, but the first part of the test are the same.

File: src/components/molecules/p6-tabs/p6-tabs.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+  styleUrl: "p6-tabs.scss",
+  shadow: true,
+})
+export class P6Tabs {
+  @Element() host!: HTMLP6TabsElement;
+
+  /**
+   * Default tab selected.
+   */
+  @Prop() default!: string;
```
Comment (vincentv) [inline src/components/molecules/p6-tabs/p6-tabs.tsx:28]: If no child is selected, the 1st child could be selected automatically. We will not need to put "default" in required

File: src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx (hunk 5 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx:19]: Updated to be the third one

File: src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx (hunk 4 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/test/p6-tabs.spec.tsx:32]: Snapshot added

File: src/components/molecules/p6-tabs/p6-tabs.stories.mdx (hunk 2 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.stories.mdx:10]: Preview added

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 7 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:28]: Done

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 3 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: That would be adding another filter just to keep a part of a condition mutualized ?

File: src/components/molecules/p6-tabs/p6-tabs.tsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  /**
+   * Return the formated tab id
+   *
+   * @param { string } id
+   */
+  private getTabId = (id: string): string => {
+    return `${id}-tab`;
+  };
```
[the remark this answers is not in the document]
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:55]: Needed because if I don't use `this` on a method instead of a property there will be an error

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 3 above)
Comment (jswale) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: For me, the test on getContent is just a more restrictive test then the one on getTabs. So yes, having 2 filter call on the getContent is not a problem for me.

If we decide to add unit test on this method, we wont need to duplicate the logic twice

What is your POV @vincentv

Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: So if I have 100 tabs open, I will have to loop 200 times ?

Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: Take into account that each time the `p6-tabs` is updated, it will re-render the tabs, so it will trigger the filter again

File: src/components/molecules/p6-tabs/p6-tabs.tsx [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    if (this.getTabs().some((tab) => tab.id === id)) {
+      this.tabSelected = this.getTabId(id);
+    }
+  }
+
+  /**
+   * Return the formated tab id
+   *
+   * @param { string } id
+   */
```
[the remark this answers is not in the document]
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:52]: I don't really understand the part `and it's on a private method.`, if the comment are not usefull it's ok, but what the point of never putting comment on a private method in the case of a complexe one ?

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 3 above)
Comment (jswale) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: Then keep the filter calls and just extract the logic of the first one into a method and call that method on both the filters

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (jswale) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:52]: Comments are good for me

File: src/components/molecules/p6-tabs/p6-tabs.tsx [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+  private handleTabSelection = (event: MouseEvent): void => {
+    event.preventDefault();
+    const { id } = event.currentTarget as HTMLAnchorElement;
+    this.tabSelected = id;
+  };
+
+  /**
+   * Get tabs from slot sttributes.
+   */
```
[the remark this answers is not in the document]
Comment (jswale) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:70]: Comments on private method are not a bad thing.

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:55]: Screenshot 2020-06-10 at 10 56 00

[the remark this answers is not in the document]
Comment (jswale) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:55]: this.id yes :)

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 3 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:73]: Added a method to test the validity of a tab

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:55]: ?

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 6 above)
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:46]: Replacing the `default` prop with a `selected` one would mean that the `p6-tabs` will no longer handle the state anymore, there will be no point to keep tabs state outside and inside the `p6-tabs`

Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:46]: Maybe it's better to just remove the method itself, I don't see a use case as the default seems to handle what we nned on the portal right now

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 3 above)
Comment (vincentv) [reply src/components/molecules/p6-tabs/p6-tabs.tsx]: > For me, the test on getContent is just a more restrictive test then the one on getTabs. So yes, having 2 filter call on the getContent is not a problem for me.
> 
> If we decide to add unit test on this method, we wont need to duplicate the logic twice
> 
> What is your POV @vincentv

I share your point of view.

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (vincentv) [reply src/components/molecules/p6-tabs/p6-tabs.tsx:50]: extract this function from the class...
add a `utils.ts` into your component folder :grin:

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (vincentv) [reply src/components/molecules/p6-tabs/p6-tabs.tsx]: They are useless if they don't provide more information than the name and typing.

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (vuongc) [reply src/components/molecules/p6-tabs/p6-tabs.tsx]: done

File: src/components/molecules/p6-tabs/p6-tabs.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (vincentv) [reply src/components/molecules/p6-tabs/p6-tabs.tsx]: On a public method, comments will be used for documentation, this is not the case for a private method.

On a private method if you need to add a comment in addition to naming and typing, the method is too complex and should preferably be refactored.

File: src/global/_variable.scss [hunk 11]
```diff
@@ -1,6 +1,13 @@
...
+
 $family-primary: "Open Sans", helvetica, arial, sans-serif;
 
 $gap-size: 5;
 
 $size-small: 0.85rem;
 $size-xlarge: 2rem;
+
+$tabs-link-active-border-bottom-color: $success;
+$tabs-link-active-color: $success;
```
Comment (vincentv) [inline src/global/_variable.scss:13]: This file is there to override the "bulma" variables. The variables added must not depend on a "bulma" variable unless they have been previously overridden in this file.

You should declare these variables in the `p6-tabs.scss` file with the suffix !default.

Comment (vuongc) [reply src/global/_variable.scss]: done
